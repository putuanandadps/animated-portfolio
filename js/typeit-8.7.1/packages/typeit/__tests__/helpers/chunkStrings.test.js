import {
  chunkStringAsHtml,
  maybeChunkStringAsHtml,
  walkElementNodes,
} from "../../src/helpers/chunkStrings";
import expandTextNodes from "../../src/helpers/expandTextNodes";

beforeEach(() => {
  document.body.innerHTML = "";
});

test("Parses normal string correctly.", () => {
  let result = chunkStringAsHtml("Hello, this is my string.");

  expect(result).toMatchSnapshot();
});

test("Parses single HTML tag.", () => {
  let result = chunkStringAsHtml(
    "Hello, this is some <strong>bold</strong> text."
  );

  expect(result).toMatchSnapshot();
});

test("Parses multiple HTML tags.", () => {
  let result = chunkStringAsHtml(
    "Hello, this is some <strong>bold</strong> text, and some <i>italicized</i> text."
  );

  expect(result).toMatchSnapshot();
});

test("Parses HTML tag at beginning of string.", () => {
  let result = chunkStringAsHtml(
    "<strong>Hello!</strong> This is some text with HTML at the beginning."
  );

  expect(result).toMatchSnapshot();
});

test("Parses HTML tag at end of string.", () => {
  let result = chunkStringAsHtml(
    "This is some text with HTML at the <em>end.</em>"
  );

  expect(result).toMatchSnapshot();
});

test("Parses HTML tag with attributes.", () => {
  let result = chunkStringAsHtml(
    'This string has an <strong class="strong-class" id="strong-id" data-whatever="data-att">element</strong> with attributes.'
  );

  expect(result).toMatchSnapshot();
});

describe("maybeChunkStringAsHtml()", () => {
  test("Should return noderized string when setting is enabled.", () => {
    expect(
      maybeChunkStringAsHtml("A <em>fancy</em> string.")
    ).toMatchSnapshot();
  });

  test("Should correctly transform non-HTML string as character objects.", () => {
    expect(
      maybeChunkStringAsHtml("A <em>fancy</em> string.", false)
    ).toMatchSnapshot();
  });
});

describe("walkElementNodes()", () => {
  test("Gathers correct number of nodes with original parents attached.", () => {
    setHTML`<h2 id="heading">A<span>B<small>C</small></span></h2>`;

    const el = expandTextNodes(document.getElementById("heading"));
    const result = walkElementNodes(el);
    const thirdNode = result[2];

    // "B" should have an original parent of the <span>.
    expect(thirdNode.originalParent).toMatchObject(
      document.querySelector("span")
    );

    // Includes each text node, and separate nodes for the parent elements themselves.
    expect(result.length).toEqual(5);
  });

  test("Reverses nodes when parameter is passed.", () => {
    setHTML`<h2 id="heading">ABC</h2>`;

    const el = expandTextNodes(document.getElementById("heading"));
    const result = walkElementNodes(el, true);

    expect(result[0].textContent).toEqual("C");
  });

  test("Excludes cursor.", () => {
    setHTML`<h2 id="heading"><span>123</span><i class="ti-cursor">|</i></h2>`;

    const el = expandTextNodes(document.getElementById("heading"));
    const result = walkElementNodes(el);
    const cursor = result.find((n) => n?.classList?.contains("ti-cursor"));

    expect(cursor).toBe(undefined);
    expect(result).toHaveLength(4);
  });
});
