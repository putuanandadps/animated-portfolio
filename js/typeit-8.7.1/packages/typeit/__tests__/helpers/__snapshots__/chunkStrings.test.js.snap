// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parses HTML tag at beginning of string. 1`] = `
[
  <strong>
    H
    e
    l
    l
    o
    !
  </strong>,
  H,
  e,
  l,
  l,
  o,
  !,
   ,
  T,
  h,
  i,
  s,
   ,
  i,
  s,
   ,
  s,
  o,
  m,
  e,
   ,
  t,
  e,
  x,
  t,
   ,
  w,
  i,
  t,
  h,
   ,
  H,
  T,
  M,
  L,
   ,
  a,
  t,
   ,
  t,
  h,
  e,
   ,
  b,
  e,
  g,
  i,
  n,
  n,
  i,
  n,
  g,
  .,
]
`;

exports[`Parses HTML tag at end of string. 1`] = `
[
  T,
  h,
  i,
  s,
   ,
  i,
  s,
   ,
  s,
  o,
  m,
  e,
   ,
  t,
  e,
  x,
  t,
   ,
  w,
  i,
  t,
  h,
   ,
  H,
  T,
  M,
  L,
   ,
  a,
  t,
   ,
  t,
  h,
  e,
   ,
  <em>
    e
    n
    d
    .
  </em>,
  e,
  n,
  d,
  .,
]
`;

exports[`Parses HTML tag with attributes. 1`] = `
[
  T,
  h,
  i,
  s,
   ,
  s,
  t,
  r,
  i,
  n,
  g,
   ,
  h,
  a,
  s,
   ,
  a,
  n,
   ,
  <strong
    class="strong-class"
    data-whatever="data-att"
    id="strong-id"
  >
    e
    l
    e
    m
    e
    n
    t
  </strong>,
  e,
  l,
  e,
  m,
  e,
  n,
  t,
   ,
  w,
  i,
  t,
  h,
   ,
  a,
  t,
  t,
  r,
  i,
  b,
  u,
  t,
  e,
  s,
  .,
]
`;

exports[`Parses multiple HTML tags. 1`] = `
[
  H,
  e,
  l,
  l,
  o,
  ,,
   ,
  t,
  h,
  i,
  s,
   ,
  i,
  s,
   ,
  s,
  o,
  m,
  e,
   ,
  <strong>
    b
    o
    l
    d
  </strong>,
  b,
  o,
  l,
  d,
   ,
  t,
  e,
  x,
  t,
  ,,
   ,
  a,
  n,
  d,
   ,
  s,
  o,
  m,
  e,
   ,
  <i>
    i
    t
    a
    l
    i
    c
    i
    z
    e
    d
  </i>,
  i,
  t,
  a,
  l,
  i,
  c,
  i,
  z,
  e,
  d,
   ,
  t,
  e,
  x,
  t,
  .,
]
`;

exports[`Parses normal string correctly. 1`] = `
[
  H,
  e,
  l,
  l,
  o,
  ,,
   ,
  t,
  h,
  i,
  s,
   ,
  i,
  s,
   ,
  m,
  y,
   ,
  s,
  t,
  r,
  i,
  n,
  g,
  .,
]
`;

exports[`Parses single HTML tag. 1`] = `
[
  H,
  e,
  l,
  l,
  o,
  ,,
   ,
  t,
  h,
  i,
  s,
   ,
  i,
  s,
   ,
  s,
  o,
  m,
  e,
   ,
  <strong>
    b
    o
    l
    d
  </strong>,
  b,
  o,
  l,
  d,
   ,
  t,
  e,
  x,
  t,
  .,
]
`;

exports[`maybeChunkStringAsHtml() Should correctly transform non-HTML string as character objects. 1`] = `
[
  A,
   ,
  &lt;,
  e,
  m,
  &gt;,
  f,
  a,
  n,
  c,
  y,
  &lt;,
  /,
  e,
  m,
  &gt;,
   ,
  s,
  t,
  r,
  i,
  n,
  g,
  .,
]
`;

exports[`maybeChunkStringAsHtml() Should return noderized string when setting is enabled. 1`] = `
[
  A,
   ,
  <em>
    f
    a
    n
    c
    y
  </em>,
  f,
  a,
  n,
  c,
  y,
   ,
  s,
  t,
  r,
  i,
  n,
  g,
  .,
]
`;
